### PR TITLE
Add missing async content loading logic to NewsOverlay

### PR DESCRIPTION
**Pull request by Game4all**
Split from #7016

Adds the missing bits for loading asynchronously and showing content on the NewsOverlay

### CODE REVIEW

File: osu.Game.Tests/Visual/Online/TestSceneNewsOverlay.cs [hunk 1]
```diff
@@ -2,22 +2,28 @@
...
             AddStep(@"Show", news.Show);
             AddStep(@"Hide", news.Hide);
 
             AddStep(@"Show front page", () => news.ShowFrontPage());
             AddStep(@"Custom article", () => news.Current.Value = "Test Article 101");
         }
+
+        private class TestNewsOverlay : NewsOverlay
+        {
+            public new void LoadAndShowChild(NewsContent content) => base.LoadAndShowChild(content);
```
Comment (peppy) [inline osu.Game.Tests/Visual/Online/TestSceneNewsOverlay.cs]: no usage of this?

Comment (Game4all) [reply osu.Game.Tests/Visual/Online/TestSceneNewsOverlay.cs]: This is being used in upcoming PR for the news overlay for testing purposes. Guess it is better to keep that for when it is being used?

Comment (peppy) [reply osu.Game.Tests/Visual/Online/TestSceneNewsOverlay.cs]: generally we avoid adding any new functions without having test coverage.

**Comment by peppy** [comment]
I'm still confused. You forced pushed this but the new method is still not used. Either make a test for it or combine it with the follow-up PR.

**Comment by Game4all** [comment]
Closing as the commit is being cherry-picked in #7192

**Pull request closed without merging**